transport/grpc/options.go

```diff
@@ -119,25 +121,35 @@ func ClientMaxSendMsgSize(clientMaxSendMsgSize int) TransportOption {
 	}
 }
 
-// ClientTLS says to use TLS with the system certificate pool on the client side.
-//
-// The default is to not use TLS.
-func ClientTLS() TransportOption {
-	return func(transportOptions *transportOptions) {
-		transportOptions.clientTLS = true
-	}
-}
-
 // InboundOption is an option for an inbound.
 type InboundOption func(*inboundOptions)
 
 func (InboundOption) grpcOption() {}
 
+// Creds returns an InboundOption that sets credentials for incoming
+// connections.
+func Creds(creds credentials.TransportCredentials) InboundOption {
+	return func(inboundOptions *inboundOptions) {
+		inboundOptions.creds = creds
+	}
+}
+
 // OutboundOption is an option for an outbound.
 type OutboundOption func(*outboundOptions)
 
 func (OutboundOption) grpcOption() {}
 
+// DialOption is an option that influences grpc.Dial.
+type DialOption func(*dialOptions)
+
+// WithTransportCredentials returns a DialOption which configures a
+// connection level security credentials (e.g., TLS/SSL).
+func WithTransportCredentials(creds credentials.TransportCredentials) DialOption {
+	return func(dialOptions *dialOptions) {
+		dialOptions.creds = creds
+	}
+}
```

Comment:
The naming here has some asymmetry: why `WithFoo` for outbounds, but `Foo` for inbounds? Same for `Creds` and `Credentials`.

Reply:
See above discussion above.
https://godoc.org/google.golang.org/grpc#Creds
https://godoc.org/google.golang.org/grpc#WithTransportCredentials

Reply:
Oh, gRPC. Ok.

Reply:
We're going with @abg's naming suggestion.